### DISCUSSION TITLE
Best way to copy database to other database connection

### DISCUSSION THREAD

**Discussion by iodev** [prisma/prisma#11936] [Q&A] [answered]
Can you have two database connections and copy one to the other, or is there a copy button for that?
We have two environments, within the same project, but want to be able to copy the production environment data to the staging environment, same schema.  
What is the best way to do that?

**Comment by nurul3101** [comment, marked as answer]
Hey @iodev 👋🏻 ,

The best way to copy data from one database server to another database server would be by creating database dump of source database server and restoring the same dump to destination database server.

Steps for **PostgreSQL**:

1. Create a Dump file of the source database.
```
pg_dump -U postgres -d source_database -f source_database.sql
```
2. Copy the dump file to the remote server.
3. Create a new database in the remote server where you want to restore the database dump:
```
CREATE DATABASE target_database;
```
4. Restore the dump file on the remote server: 
```
psql -U postgres -d target_database -f source_database.sql
```

In context of `Prisma Data Platform` I can understand that it would be a nice feature to have a copy button or some way to create new environment from an existing environment, currently we don't have it. I'll pass this feedback to our product team.

Please let me know if this solves it for you or if there are any other questions.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
